File: leetcode-python-go/791. 自定义字符串排序/main.py
from collections import defaultdict

order = "cba"
s = "abcd"

def customSortString(order: str, s: str) -> str:
    val = defaultdict(int)
    for i, ch in enumerate(order):
        val[ch] = i + 1
    
    return "".join(sorted(s, key=lambda ch: val[ch]))


print(customSortString(order, s))

File: leetcode-python-go/791. 自定义字符串排序/main.go
package main

import (
	"fmt"
	"sort"
)

func customSortString(order, s string) string {
	val := map[byte]int{}
	for i, c := range order {
		val[byte(c)] = i + 1
	}
	t := []byte(s)
	sort.Slice(t, func(i, j int) bool { return val[t[i]] < val[t[j]] })
	return string(t)
}

func main() {
	order := "cba"
	s := "abcd"
	results := customSortString(order, s)
	fmt.Println(results)
}
